pkgs/os-specific/linux/google-authenticator/default.nix

```diff
@@ -1,27 +1,28 @@
-{ stdenv, lib, fetchurl, pam, qrencode }:
+{ stdenv, lib, fetchurl, autoreconfHook, pam, qrencode }:
 
 stdenv.mkDerivation rec {
-  name = "google-authenticator-1.0";
+  name = "google-authenticator-libpam-${version}";
+  version = "1.03";
 
   src = fetchurl {
-    url = "https://google-authenticator.googlecode.com/files/libpam-${name}-source.tar.bz2";
-    sha1 = "017b7d89989f1624e360abe02d6b27a6298d285d";
+    url = "https://github.com/google/google-authenticator-libpam/archive/${version}.tar.gz";
+    sha256 = "0wb95z5v1w4sk0p7y9pbn4v95w9hrbf80vw9k2z2sgs0156ljkb7";
   };
 
-  buildInputs = [ pam ];
+  buildInputs = [ autoreconfHook pam ];
 
   preConfigure = ''
-    sed -i 's|libqrencode.so.3|${qrencode}/lib/libqrencode.so.3|' google-authenticator.c
+    sed -i "s|libqrencode.so.3|$qrencode/lib/libqrencode.so.3|" src/google-authenticator.c
```

Comment:
I like to use environment variables where possible over Nix antiquotations. Here it doesn't make a difference. However, with `version` and other values that can't be `override`n easily, it makes it easier to call `updateAttrs` since the old value of e.g. `version` isn't baked into the various shell snippets.